src/store/undoRedo/index.ts

```diff
@@ -115,15 +124,29 @@ export const actions: ActionTree<IUndoRedoState, IRootState> = {
       flows: rootState.flow,
     })
 
+    let isLastFlowStateFromPersistence = !getters.hasCurrentSnapshot
+    const currentSavedAt = getters.currentSnapshot?.modules.flows?.savedAt
+    const nextSavedAt = nextSnapshot?.modules.flows?.savedAt
+    if (currentSavedAt !== undefined && nextSavedAt !== undefined && currentSavedAt !== nextSavedAt) {
+      isLastFlowStateFromPersistence = true
+    }
+
+    // ####### force patch ##############
+    if (isLastFlowStateFromPersistence) {
+      commit('patchSnapshot', nextSnapshot)
+      return
+    }
```

Comment:
can we merge this with the `commit` below?

Reply:
I'd prefer to leave the "early return" pattern, and maybe update the existing code to something like
```
if(getters.hasFutureSnapshot) {
      commit('addSnapshot', nextSnapshot)
      return
}

const hasSpecialKeys = ...
if(changedKeys) {
      commit('addSnapshot', nextSnapshot)
      return
}

...
commit('patchSnapshot', nextSnapshot)
````

That way we don't need to way for the whole `hasDifferentKeys || getters.hasFutureSnapshot || hasSpecialKeys` to be ready before we decide
Your thoughts @object-required , @bulatgab

Reply:
To me - that would be another option, yes. Because then we would be doing a _consistent_ early return 👍